pkg/inject/sidecar_test.go

```diff
@@ -222,6 +222,22 @@ func TestInjectSidecarWithVolumeMounts(t *testing.T) {
 	assert.NotContains(t, dep.Spec.Template.Spec.Containers[1].VolumeMounts, commonVolumeMount)
 }
 
+func TestSidecarImagePullSecrets(t *testing.T) {
+
+	agentImagePullSecrets := []corev1.LocalObjectReference{{
+		Name: "agentImagePullSecret",
+	}}
+
+	jaeger := v1.NewJaeger(types.NamespacedName{Name: "TestSidecarImagePullSecrets"})
+	jaeger.Spec.Agent.ImagePullSecrets = agentImagePullSecrets
+
+	dep := dep(map[string]string{}, map[string]string{})
```

Comment:
Could you add another test, showing that a pull secret that is specified at the original `Deployment` is kept, while the new `imagePullSecret` is added as well?

Reply:
Done